compiler/jsgen.nim

```diff
@@ -1319,6 +1323,10 @@ proc genAddr(p: PProc, n: PNode, r: var TCompRes) =
     gen(p, n[0], r)
   of nkHiddenDeref:
     gen(p, n[0][0], r)
+  of nkHiddenAddr:
+    gen(p, n[0], r)
+  of nkStmtListExpr:
+    gen(p, n[^1], r)
```

Comment:
Incorrect for ``addr((echo "unrelated"; x))``. (But maybe the frontend doesn't allow this?)

Reply:
oh,  so the ast results `results = addr((echo "unrelated"; x))`, how I correct handle this ? the cgen has logic turn statement list expression to statement list

Reply:
For now ensure ``n.len == 1`` and if not, produce the previous internal error.